Create Bootstrap documentation & styles for functioning mobile nav

trackleft opened the pull request:
Closes #90

trackleft commented:
Still need to add this
https://jsfiddle.net/KyleMit/ZS4L7/

trackleft commented:
Please review JS here: https://github.com/az-digital/arizona-bootstrap/compare/master..feature/93-chris
AND
https://review.digital.arizona.edu/arizona-bootstrap/feature/93-chris/docs/0.0/components/navbar/#off-canvas-mobile-nav

mmunro-ltrr commented:
Tested on a local build: worked, but the documentation example might need a warning that the interesting behavior _only_ appears on narrow viewports.

mmunro-ltrr commented:
Currently trying to understand the changes to the Jekyll setup & how the cleanup to `.gitignore` differs from upstream Bootstrap's own cleanup a while back.

trackleft commented on package.json:
```diff
@@ -77,9 +77,11 @@
   "devDependencies": {
     "@babel/cli": "^7.10.5",
     "@babel/core": "^7.11.0",
+    "@babel/plugin-proposal-object-rest-spread": "^7.10.1",
```
Oh I thought it was being added back in your merged PR. oops

akslay commented on js/src/offcanvasmenu.js:
```diff
@@ -0,0 +1,393 @@
+  VIEWPORT_WIDTH = window.innerWidth || document.documentElement.clientWidth
+}
+
+$('.dropdown.keep-open .dropdown-toggle').on('click', function (event) {
```
Should we target `.navbar-offcanvas .dropdown` instead of all `.dropdown` elements? Since there can be `.dropdown`s used outside of the main navigation.

joeparsons commented on js/src/offcanvasmenu.js:
```diff
@@ -0,0 +1,393 @@
+const EVENT_OPENED         = `opened${EVENT_KEY}`
+const EVENT_CLOSE          = `close${EVENT_KEY}`
+const EVENT_CLOSED         = `closed${EVENT_KEY}`
+const TA_API = `click${EVENT_KEY}${DATA_API_KEY}`
```
Was this variable name intentional?  (Just wondering if it should be `DATA_API`?)

trackleft commented on js/src/offcanvasmenu.js:
```diff
@@ -0,0 +1,393 @@
+
+import $ from 'jquery'
+import Util from '../../node_modules/bootstrap/js/src/util.js'
+import { version } from '../../package.json'
```
Yes, though I think that maybe your banner.js update does this in a different way?

joeparsons commented:
Here is the diff between TWBS' collapse.js and Chris' offcanvasmenu.js file:
https://gist.github.com/joeparsons/2ac36c23f99bcbf5a39f1a51c1a324fd

trackleft commented:
OK, I think I've resolved all the things that we talked about in the Friday meeting.
[x] Removed unused .gitattribute entries
[x] Checked out scripts from master
[x] Fixed typo in comment.
[x] Removed @rollup library removed in bs upstream
[x] Removed @rollup/json library and all the usage of it within js files.

mmunro-ltrr commented:
On a local build, now seeing a problem with how the off-canvas nav option gets rendered at non-mobile sizes.
![potential_off_canvas_nav_problem](https://user-images.githubusercontent.com/5680584/91612115-9161da80-e931-11ea-84af-d57d6b6c11a4.png)
(being on-canvas, & forcing a horizontal scrollbar).

akslay commented on site/docs/0.0/components/navbar-off-canvas.md:
```diff
@@ -0,0 +1,75 @@
+    <span class="material-icons"> menu </span>
+  </button>
+</div>
+<nav class="navbar-off-canvas" id="navbarOffcanvasDemo">
```
I might be in favor of changing the `.navbar-off-canvas` class to `.navbar-offcanvas` so it matches the other classes, like `.navbar-offcanvas-header`. Just for the sake of consistency. (that was my bad)

akslay commented on site/docs/0.0/components/navbar.md:
```diff
@@ -39,7 +39,6 @@ Here's an example of all the sub-components included in a responsive light-theme
 ## Example
 
 Include example of "default" navbar (`.navbar-light`).
-The use of a `button` element is recommended for the `.dropdown-toggle` in your navbar, given that your dropdown will open on click.
```
I think it might be a mistake that all of these changes were reverted to this file. We decided to add in these instructions for `button`s (and change the corresponding examples) for the sake of semantics and accessibility; to not have `a` elements as the dropdown parent, because our dropdowns activate on click and not on hover.

akslay commented on site/docs/0.0/assets/scss/_custom-docs.scss:
```diff
@@ -109,6 +109,18 @@
+// >> Off-canvas Nav example
+@include media-breakpoint-down(md) {
+
+  .navbar-offcanvas {
```
This rule isn't getting applied since since the classes are mismatched. Once we change the class name back to `.navbar-offcanvas` instead of `.navbar-off-canvas`, this will fix the gap that's appearing to the right of the mobile nav on the Bootstrap docs site.

akslay commented:
It looks like there's a weird overlap happening with the dropdowns (this is after refreshing the page at the mobile screen size).
![Screen Shot 2020-09-11 at 11 30 04 AM](https://user-images.githubusercontent.com/52931751/92960318-43b19b80-f422-11ea-96f6-97fd29763b62.png)

akslay commented on site/docs/0.0/components/navbar-off-canvas.md:
```diff
@@ -0,0 +1,75 @@
+    <li class="nav-item nav-item-parent active">
+      <a class="nav-link" href="#">Home <span class="sr-only">(current)</span></a>
+    </li>
+    <li class="nav-item">
```
This item needs the `.nav-item-parent` class to receive the uppercase styling for parent items.

akslay commented on site/docs/0.0/components/navbar-off-canvas.md:
```diff
@@ -0,0 +1,75 @@
+        <span class="material-icons">home</span>
+        <span>HOME</span>
+      </a>
+      <button id="navbarOffcanvasDemoClose" data-toggle="offcanvas" type="button" data-target="#navbarOffcanvasDemo" aria-controls="navbarOffcanvasDemo" class="d-flex flex-column align-items-center btn btn-default">
```
Change `.btn.btn-default` to `.btn.btn-red` (`.btn-default` is deprecated & backwards compatible, but probably don't want to include it in examples since we want to discourage its use).

akslay commented on site/docs/0.0/components/navbar-off-canvas.md:
```diff
@@ -0,0 +1,75 @@
+</div>
+<nav class="navbar-off-canvas" id="navbarOffcanvasDemo">
+  <div class="navbar-offcanvas-header">
+    <div class="bg-chili py-1 px-3 d-flex justify-content-between align-items-center">
```
Remove `py-1 px-3` to fixing spacing issue (to match mockup).

akslay commented on site/docs/0.0/components/navbar-off-canvas.md:
```diff
@@ -0,0 +1,75 @@
+        <span>CLOSE</span>
+      </button>
+    </div>
+    <form class="navbar-offcanvas-search bg-white pr-3">
```
Remove `pr-3` to fix spacing (to match mockup).

akslay commented on scss/_navbar-off-canvas.scss:
```diff
@@ -0,0 +1,295 @@
+  animation: .3s linear fadein;
+}
+
+.navbar-off-canvas {
```
I also have this entire block nested inside a media breakpoint so that these styles are only being applied at the mobile size. I've attached a .txt file with the most recent SCSS I have.
[_navbar-offcanvas.txt](https://github.com/az-digital/arizona-bootstrap/files/5211036/_navbar-offcanvas.txt)

akslay commented on scss/_custom-styles.scss:
```diff
@@ -109,6 +109,11 @@
 }
+// >>> Material design icons.
+.material-icons {
+  font-size: inherit;
```
It looks like this style is preventing utility classes (like `.display-3`) from working as expected; it's not allowing the class to change the font size.

akslay commented on site/docs/0.0/components/navbar-off-canvas.md:
```diff
@@ -0,0 +1,75 @@
+        <a class="dropdown-item" href="#">Something else here</a>
+      </div>
+    </li>
+    <li class="nav-item">
```
This one also needs `.nav-item-parent` for the uppercase styling

mmunro-ltrr reviewed (approved):
Previous approval was in the distant past, so wasn't really relevant.
